docs/book/basics/simple_controller.md

```diff
@@ -148,79 +148,78 @@ on the Deployment once the ContainerSet is deleted.
+  reconcile.Result, error) {
     // Read the ContainerSet
-	cs := &workloadv1beta1.ContainerSet{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, cs)
+  cs := &workloadv1beta1.ContainerSet{}
+  err := r.client.Get(context.TODO(), request.NamespacedName, cs)
     
     // Handle deleted or error case
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Not found.  Don't worry about cleaning up Deployments,
-			// GC will handle it.
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
-	}
+  if err != nil {
+    if errors.IsNotFound(err) {
+      // Not found.  Don't worry about cleaning up Deployments,
+      // GC will handle it.
+      return reconcile.Result{}, nil
+    }
+    // Error reading the object - requeue the request.
+    return reconcile.Result{}, err
+  }
 
     // Calculate the expected Deployment Spec
-	spec := getDeployment(request)
+  spec := getDeploymentSpec(request)
 
     // Read the Deployment
-	dep := &v1.Deployment{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, dep)
-
-    // If not found, create it
-    if errors.IsNotFound(err) {
-        dep = &appsv1.Deployment{Spec: spec}
-		dep.Name = request.Name
-		dep.Namespace = request.Namespace
-    	if err := controllerutil.SetControllerReference(cs, deploy, r.scheme); err != nil {
-	    	return reconcile.Result{}, err
-	    }
-		if err := r.Create(context.TODO(), dep); err != nil {
-	    	return reconcile.Result{}, err
-		}
-		return reconcile.Result{}, nil
-	}
+  dep := &appsv1.Deployment{}
+  err := r.client.Get(context.TODO(), request.NamespacedName, dep)
+
+    // If not found, create it 
+  if errors.IsNotFound(err) {
+    dep = &appsv1.Deployment{Spec: spec}
+    dep.Name = request.Name
+    dep.Namespace = request.Namespace
+    if err := controllerutil.SetControllerReference(cs, deploy, r.scheme); err != nil {
+      return reconcile.Result{}, err
+    }
+    if err := r.Create(context.TODO(), dep); err != nil {
+      return reconcile.Result{}, err
+    }
+    return reconcile.Result{}, nil
+  }
 
     // If found, update it
     image := dep.Spec.Template.Spec.Containers[0].Image
     replicas := *dep.Spec.Replicas
     if replicas == cs.Spec.Replicas && image == cs.Spec.Image {
-        return reconcile.Result{}, nil
+      return reconcile.Result{}, nil
     }
     dep.Spec.Replicas = &cs.Spec.Replicas
     dep.Spec.Template.Spec.Containers[0].Image = cs.Spec.Image
     if err := r.Update(context.TODO(), dep); err != nil {
-        return reconcile.Result{}, err
+      return reconcile.Result{}, err
     }
     
     return reconcile.Result{}, nil
 }
 
-func getDeployment(request reconcile.Request) *v1.Deployment {
-	return &appsv1.DeploymentSpec{
-        Selector: &metav1.LabelSelector{
-            MatchLabels: map[string]string{
-                "container-set": request.Name},
+func getDeploymentSpec(request reconcile.Request) *appsv1.Deployment {
```

Comment:
Should be `*appsv1.DeploymentSpec` in the func signature.

Reply:
Good catch